src/main/kotlin/org/rust/ide/structure/RsStructureViewModel.kt

```diff
@@ -31,8 +33,26 @@ class RsStructureViewModel(editor: Editor?, file: RsFileBase) :
 
     init {
         withSuitableClasses(
-            RsNamedElement::class.java,
-            RsImplItem::class.java
+            RsImplItem::class.java,
+            RsMacro::class.java,
+            RsMacro2::class.java,
+            RsFunction::class.java,
+
+            /**
+             * Keep in sync with [RsBreadcrumbsInfoProvider.RsNamedHandler].
+             * See also [RsNavBarModelExtension].
+             */
+            RsConstant::class.java,
+            RsEnumItem::class.java,
+            RsEnumVariant::class.java,
+            RsExternCrateItem::class.java,
+            RsModDeclItem::class.java,
+            RsModItem::class.java,
+            RsNamedFieldDecl::class.java,
+            RsStructItem::class.java,
+            RsTraitAlias::class.java,
+            RsTraitItem::class.java,
+            RsTypeAlias::class.java,
```

Comment:
Here are list of classes which were previously included (as subclasses of `RsNamedElement`), but are not excluded:
```
RsFile
RsCodeFragment
RsAlias
RsConstParameter
RsLabelDecl
RsLifetime
RsLifetimeParameter
RsMacroBinding
RsPatBinding
RsSelfParameter
RsTypeParameter
```

Reply:
If we want to sync the classes supported by `RsBreadcrumbsInfoProvider` and `RsStructureViewModel`, we can extract a list of these classes and reuse it in both places like this:
```Kotlin
val SUITABLE_CLASSES: Array<Class<out RsElement>> = arrayOf(
    RsModItem::class.java,
    RsStructOrEnumItemElement::class.java,
    RsTraitItem::class.java,
    RsConstant::class.java,
    RsTypeAlias::class.java,
    RsTraitAlias::class.java,
    RsNamedFieldDecl::class.java,
    RsEnumVariant::class.java,
    RsModDeclItem::class.java,
    RsExternCrateItem::class.java
)

// RsNamedHandler
override fun accepts(e: PsiElement): Boolean = SUITABLE_CLASSES.any { it.isInstance(e) }

// RsStructureViewModel
withSuitableClasses(
    // ...
    *SUITABLE_CLASSES
)
```
This can make it easier to keep them synced.

Also, since these two classes share some logic, I think we can move `RsBreadcrumbsInfoProvider` to the `org.rust.ide.structure` package as a more relevant place.